### ISSUE TITLE
Question: I want to use this for game clips, does it have some simple editing options?

### ISSUE THREAD

**Issue by sideh** [mediacms-io/mediacms#1032]
**Describe the feature you'd like**
I would like to upload a video that I recorded from a game but then clip it size. By default when I press my record button, it saves the last 5 mintues of video. I want to then upload that to this and then edit it, nothing fancy but be able to drag a slider to a minute or 2.

**Describe alternatives you've considered**
Doing this through YouTube but with the shareing/visibility set to private (Not sure if that is even an option, just an assumption) or even just slowly editing the clips one day in a video editing software.

**Additional context**
Just want a nice easy way to share clips to friends with out needing YouTube, unless it is the easiest way of doing that. Happy for alternatives/ideas. Or if this can do that then fantastic.

**Comment by azataiot** [comment]
Editing video is not the main purpose of this project. The main purpose of this project is to let you able to SERVE your media files which you have already done your editing properly.

**Comment by sideh** [comment]
No worries, thanks for clarifying.

**Comment by Zero3** [comment]
@sideh did you ever find another solution? I have the same use case as you, and was looking into whether MediaCMS could support it as well. Basically some kind of video gallery that can be self-hosted, has simple clipping/editing functionality and allows the user to grab a direct URL to the resulting video file for sharing elsewhere. Bonus points for a simple account system and automatic reencoding according to some preset.

**Comment by mgogoulos** [comment]
That was just in time. There's a trim editor just deployed on https://demo.mediacms.io/ , feel free to upload a video and experiment with trimming. This will find it's way soon to the main version!

**Comment by Zero3** [comment]
@mgogoulos Interesting. Thanks for letting me know! I tried it out, and it is not bad :). It is not _great_ for my particular use case either, though. Some issues:

- There is too much waiting time when uploading just a single video. One cannot begin the trim process until the video has been both uploaded and reencoded (in at least one resolution?), which means waiting a minute or two for the upload (a typical ~60 second video capture of a game is a couple of hundred MBs) and then quite a few minutes for the reencoding as well. This quickly adds up to be quite inconvenient when frequently making clips. I would prefer being able to trim the original file instantly after upload, or ideally, before/during upload, to not have to wait for the upload either. I am thinking this could be implemented by having the preview work on the local copy of the file until it has been uploaded.

- To make things even more inconvenient, the video page keeps saying the video is being reencoded, even after it is actually done, until one manually refreshes the page.

- The trimming is not part of the upload flow, but is kind of hidden away and has to be activated manually. In my use case, one would always want to trim the ~60 second uploaded clip before sharing it.

- I find the trimming UI is a bit confusing. I think there is room for some UX improvements.

- I also noticed a couple of minor bugs, but that is the least of my worries really.

**Comment by Zero3** [comment]
Another piece of feedback on top of the above:

- When uploading a second video, it starts out unavailable for even longer than the first one (seemingly for over 10 minutes?), stuck in the "Media encoding is pending" state. I am guessing that the server is still busy reencoding the first video in various resolutions, and thereby blocking work on all other videos until it is done. And that was just with a 60 second video, so I assume it will be much worse with longer videos. That is a dealbreaker for my use case, where a bunch of friends are playing games and chatting together live and want to share clips with each other.

**Comment by mgogoulos** [comment]
hey thanks for checking this out and sharing thoughts. Unfortunately it's true that the video has to be encoded in order for it to be trimmed/edited. It cannot work before uploading. Maybe we could add a smaller resolution so that the video gets instantly cut so this will minimize the time. We'll see!

**Comment by mgogoulos** [comment]
There have been some improvements on the workflow and the functionality offered by the editor. You can now start trimming immediately after upload, without waiting for the transcoding to be completed. @Zero3 it would be great if you can re-do the previous experiments you did and see what your feedback is - on https://demo.mediacms.io/

**Comment by Zero3** [comment]
Hi @mgogoulos

Sure! I gave it a go. There is still quite a few flow/UX challenges that makes MediaCMS less than ideal for my use case. I will include a list of issues below, approximately ordered by the place in the flow I ran into them. Keep in mind that this is just feedback based on how I ideally would want the flow and functionality to be for my use case, and may or may not be what MediaCMS should do in general. So take it or leave it, so to say :).

1. Before one can even visit the video (which is needed to access the trimming tool?), one still needs to wait for both the actual upload as well as a "processing" stage to be completed on the upload screen. As suggested above, one would ideally be able to trim videos directly from this view, based on a local in-memory version of the file. The trim details could then be submitted to MediaCMS and the actual trim be performed in the background on the server after the upload has been completed.

2. Regardless of point 1. above, I would expect MediaCMS to ask whether I want to trim the video before it is published, as part of the upload flow. It is a bit cumbersome to get started with the trimming when one first have to wait, then navigate to the video, then find the edit button and finally find the trim tab.

3. After upload, the video has a distracting banner saying "Media encoding hasn't started yet. Attempting to show the original video file". I would rather not see this, as I see it as completely fine by me to see the original video, which is what I would prefer anyway. So I would never want this message to be shown. Any subsequently available lower quality reencodings would just be a hidden "bonus".

4. The video does not actually work while the banner above is displayed. It cannot be played, at least when I tested it.

5. Similarly to point 4, the trim functionality does not work until a while after upload. I see a black unplayable video with unresponsive trimming controls.

6. After a while, trimming becomes possible, but only based on a very low quality version of the video. I would rather trim based on the original high quality version.

7. After trimming, a message with a countdown to a redirect is shown. It is quite broken for me, as the browser will ask "do you really wanna leave this page" both if I wait out the timer and if I click to skip it. I would prefer this popup to not exist at all, and automatically be redirected instantly.

8. When playing a video, it becomes grayed out and control buttons are displayed on top of it. This stays for several seconds. If I move my mouse, it reappears. Only moving the mouse out of the video makes it stay away. This is quite annoying, especially for small sub-minute clips, where a large fraction of the clip will be covered by it.

9. The problem with the re-encoding of a single clip blocking all other uploads is still present. It takes a *long* time to be able to view or trim any videos besides the first one, even when the uploaded videos are quite short.

Hopefully some of the feedback above will be useful for you if you want to make MediaCMS more suitable for use cases like mine :).
